InputTextarea/InputText: Typescript incompatible ref type

### Describe the bug

When you are using `InputTextarea` component and want to access the DOM Element using the reference - it indicates the wrong type.

### Expected behavior

The `HTMLTextAreaElement` type should be used for ref.

### Reproducer

https://stackblitz.com/edit/vitejs-vite-6zkqc4?file=package.json,src%2FApp.tsx

### PrimeReact version

8.7.2

### React version

18.x

### Language

TypeScript

### Build / Runtime

Vite

### Browser(s)

Chrome 106

OK this is now causing other issue it hthe TypeScript checking...

```
Error: components/lib/inputmask/inputmask.d.ts(42,26): error TS274[9](https://github.com/primefaces/primereact/actions/runs/3462195764/jobs/5780762361#step:7:10): 'InputText' refers to a value, but is being used as a type here. Did you mean 'typeof InputText'?
Error: components/lib/inputnumber/inputnumber.d.ts(76,24): error TS2749: 'InputText' refers to a value, but is being used as a type here. Did you mean 'typeof InputText'?
Error: components/lib/mention/mention.d.ts(62,24): error TS2749: 'InputTextarea' refers to a value, but is being used as a type here. Did you mean 'typeof InputTextarea'?
Error: components/lib/password/password.d.ts(56,24): error TS2749: 'InputText' refers to a value, but is being used as a type here. Did you mean 'typeof InputText'?
```

--- a/components/lib/inputmask/inputmask.d.ts
+++ b/components/lib/inputmask/inputmask.d.ts
@@ -39,5 +39,5 @@ export interface InputMaskProps extends Omit<InputTextProps, 'onChange'> {
 }
 
 export declare class InputMask extends React.Component<InputMaskProps, any> {
-    public getElement(): InputText;
+    public getElement(): typeof InputText;
 }

--- a/components/lib/inputnumber/inputnumber.d.ts
+++ b/components/lib/inputnumber/inputnumber.d.ts
@@ -73,5 +73,5 @@ export interface InputNumberProps extends Omit<React.DetailedHTMLProps<React.HTM
 export declare class InputNumber extends React.Component<InputNumberProps, any> {
     public getFormatter(): any;
     public getElement(): HTMLSpanElement;
-    public getInput(): InputText;
+    public getInput(): typeof InputText;
 }

--- a/components/lib/inputtext/inputtext.d.ts
+++ b/components/lib/inputtext/inputtext.d.ts
@@ -11,6 +11,4 @@ export interface InputTextProps extends Omit<React.DetailedHTMLProps<React.Input
     onInput?(event: React.FormEvent<HTMLInputElement>, validatePattern: boolean): void;
 }
 
-export declare class InputText extends React.Component<InputTextProps, any> {
-    //value?: string;
-}
+export declare const InputText: React.ForwardRefExoticComponent<InputTextProps & React.RefAttributes<HTMLInputElement>>;

--- a/components/lib/inputtextarea/inputtextarea.d.ts
+++ b/components/lib/inputtextarea/inputtextarea.d.ts
@@ -10,6 +10,4 @@ export interface InputTextareaProps extends Omit<React.DetailedHTMLProps<React.T
     tooltipOptions?: TooltipOptions;
 }
 
-export declare class InputTextarea extends React.Component<InputTextareaProps, any> {
-    value?: string;
-}
+export declare const InputTextarea: React.ForwardRefExoticComponent<InputTextareaProps & React.RefAttributes<HTMLTextAreaElement>>;

--- a/components/lib/mention/mention.d.ts
+++ b/components/lib/mention/mention.d.ts
@@ -59,6 +59,6 @@ export interface MentionProps extends Omit<React.DetailedHTMLProps<React.Textare
 
 export declare class Mention extends React.Component<MentionProps, any> {
     public getElement(): HTMLDivElement;
-    public getInput(): InputTextarea;
+    public getInput(): typeof InputTextarea;
     public getOverlay(): HTMLElement;
 }

--- a/components/lib/password/password.d.ts
+++ b/components/lib/password/password.d.ts
@@ -53,6 +53,6 @@ export interface PasswordProps extends Omit<React.DetailedHTMLProps<React.InputH
 
 export declare class Password extends React.Component<PasswordProps, any> {
     public getElement(): HTMLDivElement;
-    public getInput(): InputText;
+    public getInput(): typeof InputText;
     public getOverlay(): HTMLElement;
 }

